### ISSUE TITLE
Tokenizer issue with Cerebras' BTLM 3B

### ISSUE THREAD

**Issue by AlpinDale** [axolotl-ai-cloud/axolotl#438]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Attempting to tokenize a `oasst` style dataset with BTLM's `GPT2Tokenizer` results in the following warning being displayed during tokenization:

```
[WARNING] [axolotl._tokenize:78] [PID:3175928] Tokenizer result is empty. You may want to audit your dataset.
```

Ignoring this will lead to:
```sh
Traceback (most recent call last):                                                                                                                                                                │
  File "/home/alpin/miniconda3/envs/axolotl/lib/python3.10/site-packages/multiprocess/pool.py", line 125, in worker                                                                               │
    result = (True, func(*args, **kwds))                                                                                                                                                          │
  File "/home/alpin/miniconda3/envs/axolotl/lib/python3.10/site-packages/datasets/utils/py_utils.py", line 1347, in _write_generator_to_queue                                                     │
    for i, result in enumerate(func(**kwargs)):                                                                                                                                                   │
  File "/home/alpin/miniconda3/envs/axolotl/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 3450, in _map_single                                                                    │
    example = apply_function_on_filtered_inputs(example, i, offset=offset)                                                                                                                        │
  File "/home/alpin/miniconda3/envs/axolotl/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 3353, in apply_function_on_filtered_inputs                                              │
    processed_inputs = function(*fn_args, *additional_args, **fn_kwargs)                                                                                                                          │
  File "/home/alpin/axolotl/src/axolotl/prompt_tokenizers.py", line 125, in tokenize_prompt                                                                                                       │
    tokenized_res_prompt = self._tokenize(                                                                                                                                                        │
  File "/home/alpin/axolotl/src/axolotl/prompt_tokenizers.py", line 88, in _tokenize                                                                                                              │
    if result["input_ids"][0] == self.tokenizer.bos_token_id and strip_bos_token:                                                                                                                 │
IndexError: list index out of range
```

Along with a thousand more lines of error.

### Current behaviour

Apparently works with Alpaca format.

### Steps to reproduce

Add this to the config YAML:

```yaml
base_model: cerebras/btlm-3b-8k-base
base_model_config: cerebras/btlm-3b-8k-base
model_type: AutoModelForCausalLM
tokenizer_type: GPT2Tokenizer
trust_remote_code: true
datasets:
  - path: /home/data/axolotl/data
    type: oasst
```

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10.12

### axolotl branch-commit

commit f733d0f31e545f177f310d14a6366bab729dc16e (HEAD -> main, origin/main, origin/HEAD)

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by winglian** [comment]
@AlpinDale fixed by #442

**Comment by ehartford** [comment]
Can't wait to try it out!
